Adds history for Fortran-lang Org (#238)
fortran-lang/webpage: source/history.md

# Fortran-lang history

## Creation

> The Fortran-lang project was started at the beginning of 2020 by Ondřej Čertík
> and Milan Curcic
> ([[1]](https://ondrejcertik.com/blog/2021/03/resurrecting-fortran/),
> [[2]](https://medium.com/modern-fortran/first-year-of-fortran-lang-d8796bfa0067)).
> The effort built upon work of several members of the Fortran Standards
> Committee that built a GitHub proposals repository to track proposals to the
> Standards Committee, which lead to the creation of the Fortran Standard Library
> (`stdlib`), the Fortran Package Manager (`fpm`) and the `fortran-lang.org`
> website. The webpage quickly become the top result in major search engines for
> "Fortran". We created a Fortran Discourse where the whole Fortran community can
> meet and discuss, make announcements, ask for help, etc. The Fortran-lang
> effort was joined by hundreds of contributors. At the end of 2022 the
> Fortran-lang project was accepted under NumFOCUS for fiscal sponsorship.

Reviewer: ```suggestion
The effort to build a new community around Fortran started at the beginning of 2020
and was initially let by Ondřej Čertík [[1]][blog-ondrej] and Milan Curcic [[2]][blog-milan].
Starting in several discussions around ambitious proposals for the Fortran Standards
Committee, the Fortran Standard Library (`stdlib`) [[3]][issue-stdlib], the Fortran
Package Manager (`fpm`) [[4]][issue-fpm] and the [`fortran-lang.org`][webpage-orig]
webpage with new logo [[5]][issue-logo] were created.
With the new webpage and projects attracting more contributors, the Fortran-lang discourse
was created to provide a place for general discussions around all the things Fortran,
announcing newly started projects, getting help, etc.
Since its creation the Fortran-lang community was joined by hundreds of contributors.

[blog-ondrej]: https://ondrejcertik.com/blog/2021/03/resurrecting-fortran/
[blog-milan]: https://medium.com/modern-fortran/first-year-of-fortran-lang-d8796bfa0067
[issue-fpm]: https://github.com/j3-fortran/fortran_proposals/issues/55
[issue-stdlib]: https://github.com/j3-fortran/fortran_proposals/issues/104
[issue-logo]: https://github.com/j3-fortran/fortran_proposals/issues/47
[webpage-orig]: https://web.archive.org/web/20200504000648/https://fortran-lang.org/
```

Reviewer 2: initially let -> initially led ?

Reviewer: Sorry for my bad English :sweat_smile:

<!-- Migration date: 22-04-2022 -->

Modern Fortran for Visual Studio Code is feature-rich Visual extension
for Fortran originally created by
Miguel Carvajal and maintained by Giannis Nikiteas.
The extension migrated in early 2022 to the Fortran-lang organization
to improve integration with other Fortran-lang projects such as `fpm`
and allow for easier community collaboration.

Find out more about Modern Fortran for Visual Studio Code on
[VS Code Marketplace](https://marketplace.visualstudio.com/items?itemName=fortran-lang.linter-gfortran)
and on [GitHub](httos://github.com/fortran-lang/vscode-fortran-support).

### fortls

<!-- Migration date: 28-11-2022 -->

`fortls` is a Fortran language server created by Giannis Nikiteas
in late 2019. The project started as a fork of the
archived project `fortran-language-server` but quickly diverged from it
both in terms of features and codebase. The project migrated to the
Fortran-lang organization in late 2022 to allow for easier community
collaboration.

Find out more about `fortls` on our [website](https://fortls.fortran-lang.org/)
and on [GitHub](https://github.com/fortran-lang/fortls).

## Fortran Publication

> ## Conferences and Events

Reviewer 3: I think this is a better format, easier to quickly get a grasp of the timeline. I also de-personalized specific contributions to de-emphasize the perception of ownership. Please add anything that I missed.

```suggestion
# Fortran-lang History

* **October 2019**: J3 members created the [j3-fortran/fortran_proposals](https://github.com/j3-fortran/fortran_proposals) to bridge the gap between J3 and the broader Fortran community. Many people, J3 members and non-members alike, contributed proposals, requests for improvement, and discussions.
* **December 2019**: We created the incubator repository for the Fortran Standard Library ([fortran-lang/stdlib](https://github.com/fortran-lang/stdlib)), and the [Fortran-lang GitHub organization](https://github.com/fortran-lang) to host the Fortran stdlib repository. We also created the [Fortran-lang Twitter account](https://twitter.com/fortranlang) and the incubator repository for the [Fortran Package Manager (fpm)](https://github.com/fortran-lang/fpm) and its first minimal prototype in Rust.
* **February 2020**: We began work on the next implementation of the fpm prototype in Haskell.
* **April 2020**: We created the [Fortran-lang website](https://fortran-lang.org) and presented efforts on the [Open Source Directions podcast](https://www.youtube.com/watch?v=2NiS2tdDO_4).
* **May 2020**: We created the [Fortran Discourse](https://fortran-lang.discourse.group/).
* **July 2020**: We presented our efforts at the first international Fortran conference ([FortranCon 2020](https://www.youtube.com/playlist?list=PLeKbr7eYHjt77h90hDVC-vGzrWmvDhCAf). We also began work on the Fortran implementation of fpm.
* **November 2020**: We began work on the Fortran implementation of fpm.
* **February 2021**: Fortran-lang accepted into the Google Summer of Code 2021 program.
* **March 2021**: [test-drive](https://github.com/fortran-lang/test-drive) framework migrated and maintained at Fortran-lang.
* **July 2021**: Modernized [fftpack](https://github.com/fortran-lang/fftpack) and [minpack](https://github.com/fortran-lang/minpack) packages migrated and maintained at Fortran-lang.
* **September 2021**: We presented our efforts at [FortranCon 2021](https://www.youtube.com/watch?v=g7ECnr_Q7Ck&list=PLeKbr7eYHjt5UaV9zQtY24oEbne9_uFni). We also published the [Toward Modern Fortran Tooling and a Thriving Developer
Community](https://arxiv.org/pdf/2109.07382.pdf) paper on arXiv.
* **November 2021**: We presented fpm at the [PackagingCon 2021](https://www.youtube.com/watch?v=YG8zEM1lAVM).
* **February 2022**: Fortran-lang accepted into the Google Summer of Code 2022 program.
* **March 2022**: [The State of Fortran](https://ieeexplore.ieee.org/stamp/stamp.jsp?arnumber=9736688) paper published in IEEE CiSE.
* **April 2022**: Modern Fortran for Visual Studio Code ([VS Code Marketplace](https://marketplace.visualstudio.com/items?itemName=fortran-lang.linter-gfortran), 
[GitHub](httos://github.com/fortran-lang/vscode-fortran-support)) joined Fortran-lang.
* **November 2022**: Fortran-lang applied to become a member project of NumFOCUS. [Fortran Language Server (fortls)](https://fortls.fortran-lang.org/) joined Fortran-lang.
* **February 2023**: Fortran-lang accepted into the Google Summer of Code 2023 program.

Reviewer 2: The format is better.

> I also de-personalized specific contributions to de-emphasize the perception of ownership.

I think it's important to give credit to people who pushed and worked on these things. We should add people back in, for each item.

Reviewer 3: I don't think that's a good idea here for a few reasons but will support it if others want it. @gnikit @awvwgk @LKedward what do you think?

Author: I tend to agree with @milancurcic, at least for vscode-modern-fortran and fortls, I don't think names should be mentioned. I don't really know what I was thinking when I added them, I guess I was in a rush to get something in the PR.

Reviewer 4: In general I support de-personalization, but in the context of a historical record it doesn't make much sense. Both the original authors and maintainers can be looked up if necessary, and so sweeping them under the rug detracts from the point of such a record. I suppose a disclaimer could be added that the records are purely historical and that the community is flourishing with many contributors. Most (if not all) project histories have specific names, and do not try to pull community efforts in until later.

Reviewer 2: I agree with @HaoZeke. For example the sentence "J3 members created" would lead me to believe that J3 have started the effort. I wanted them to start the effort, but I couldn't convince them to do it, I essentially had to do lead the effort myself. Consequently I do not find this description of the history accurate.

Reviewer 2: @gnikit actually not, this PR is specifically to address NumFOCUS' request "the inclusion of a short history of the project on your website.", so indeed a historical record. So that we can make progress with NumFOCUS acceptance.

Reviewer 3: @HaoZeke, to sweep under the rug means to intentionally hide a problem so others don't see it. I have no reason to do that. I wish you'd ask me questions before assuming.

@certik why don't you give it a shot? The first version of my write-up had creator attributions but I soon ran into problems with it and removed all names. To give you a heads up, if you credit original creators of projects, most were created either by you or by me, and that's a problematic image to portray, as we have discussed before. If you try to credit everyone fairly, then you end up with a list of 200+ names, and that's simply not the purpose of this document.

Reviewer 2: @milancurcic I did give it a shot and wrote up the initial version. Then @gnikit took it and further improved it. I approved his version as well, that's fine with me to merge.

One way forward is to do both. We can have a history page with more prose. Then we can have a shorter "milestones" version along the lines you wrote and we don't have to have names there.

Reviewer 3: Thanks, Ondrej. There seems to be a misunderstanding here. Your initial version only mentions you and I creating Fortran-lang, but doesn't give credit to creators of other projects. Format aside, I think this version was good, except that I only prefer our names not being explicitly mentioned. Then I understood your response as wanting to credit everybody who pushed each project. I think that's difficult to do and better to avoid here.

I don't think the question of format is the issue here. We just need to decide what to do with attribution. I don't think mentioning just you and me is a good solution because it omits many other people who contributed significantly more than either of us across the projects.

I think this doc, just like Fortran-lang, is about the community and not any single individual, and our writing should reflect that spirit.

Reviewer 2: I agree it's about community. It's a fine balance between giving attributions, but not too much.

announcing newly started projects, getting help, etc.
Since its creation the Fortran-lang community was joined by hundreds of contributors.

[blog-ondrej]: https://ondrejcertik.com/blog/2021/03/resurrecting-fortran/
[blog-milan]: https://medium.com/modern-fortran/first-year-of-fortran-lang-d8796bfa0067
[issue-fpm]: https://github.com/j3-fortran/fortran_proposals/issues/55
[issue-stdlib]: https://github.com/j3-fortran/fortran_proposals/issues/104
[issue-logo]: https://github.com/j3-fortran/fortran_proposals/issues/47
[webpage-orig]: https://web.archive.org/web/20200504000648/https://fortran-lang.org/

## Incubator J3 repo

## The Fortran Standard Library `stdlib`

Created in the late 2019 the Fortran Standard Library provides
a community driven and agreed upon de facto "standard" library for Fortran.
The library is a collection of modules that provide a wide range of
functionality, including: containers, sorting, searching, linear algebra,
unities, fast Fourier transforms, and more.

Find out more about `stdlib` on our [website](https://stdlib.fortran-lang.org/)
and on [GitHub](https://github.com/fortran-lang/stdlib).

## The Fortran Package Manager `fpm`

> 

Author: ```suggestion

Fortran Package Manager (fpm) is a package manager and build system for Fortran.
Its key goal is to improve the user experience of Fortran programmers.
It does so by making it easier to build your Fortran program or library, run the
executables, tests, and examples, and distribute it as a dependency to other
Fortran projects.
Fpm's user interface is modeled after [Rust's Cargo](https://doc.rust-lang.org/cargo/),
so if you're familiar with that tool, you will feel at home with fpm.
Fpm's long term vision is to nurture and grow the ecosystem of modern Fortran
applications and libraries.
```

### test-drive

### vscode-fortran-support

<!-- Migration date: 22-04-2022 -->

Modern Fortran for Visual Studio Code is feature-rich Visual extension
for Fortran originally created by
Miguel Carvajal and maintained by Giannis Nikiteas.
The extension migrated in early 2022 to the Fortran-lang organization
to improve integration with other Fortran-lang projects such as `fpm`
and allow for easier community collaboration.

Find out more about Modern Fortran for Visual Studio Code on
[VS Code Marketplace](https://marketplace.visualstudio.com/items?itemName=fortran-lang.linter-gfortran)
and on [GitHub](httos://github.com/fortran-lang/vscode-fortran-support).

### fortls

<!-- Migration date: 28-11-2022 -->

`fortls` is a Fortran language server created by Giannis Nikiteas
in late 2019. The project started as a fork of the
archived project `fortran-language-server` but quickly diverged from it
both in terms of features and codebase. The project migrated to the
Fortran-lang organization in late 2022 to allow for easier community
collaboration.

Find out more about `fortls` on our [website](https://fortls.fortran-lang.org/)
and on [GitHub](https://github.com/fortran-lang/fortls).

> ## Fortran Publication
> 

Author: ```suggestion
## Fortran Publication

In April 2022, members of the Fortran-lang community published a paper in
Computing in Science & Engineering titled [_The State of Fortran_][pub-state-of-fortran]
providing an overview of the language and its ecosystem.
The paper focused on the ongoing efforts
of developing the Fortran standard library ([`stdlib`][webpage-stdlib])
and the Fortran package manager ([`fpm`][webpage-fpm]),
fostering a friendly and welcoming community online ([`Fortran-lang Discourse`][discourse]),
improving compiler support and language feature development.

[pub-state-of-fortran]: https://ieeexplore.ieee.org/document/9736688
[discourse]: https://fortran-lang.discourse.group/
[webpage-fpm]: https://fpm.fortran-lang.org/
[webpage-stdlib]: https://stdlib.fortran-lang.org/
```

in late 2019. The project started as a fork of the
archived project `fortran-language-server` but quickly diverged from it
both in terms of features and codebase. The project migrated to the
Fortran-lang organization in late 2022 to allow for easier community
collaboration.

Find out more about `fortls` on our [website](https://fortls.fortran-lang.org/)
and on [GitHub](https://github.com/fortran-lang/fortls).

## Fortran Publication

In April 2022, members of the Fortran-lang community published a paper in
Computing in Science & Engineering titled [_The State of Fortran_][pub-state-of-fortran]
providing an overview of the language and its ecosystem.
The paper focused on the ongoing efforts
of developing the Fortran standard library ([`stdlib`][webpage-stdlib])
and the Fortran package manager ([`fpm`][webpage-fpm]),
fostering a friendly and welcoming community online ([`Fortran-lang Discourse`][discourse]),
improving compiler support and language feature development.

[pub-state-of-fortran]: https://ieeexplore.ieee.org/document/9736688
[discourse]: https://fortran-lang.discourse.group/
[webpage-fpm]: https://fpm.fortran-lang.org/
[webpage-stdlib]: https://stdlib.fortran-lang.org/

> ## Conferences and Events

Author: ```suggestion

## Conferences and Events
```
